bitnami/etcd/templates/servicemonitor.yaml

```diff
@@ -13,15 +13,28 @@ metadata:
 spec:
   selector:
     matchLabels: {{- include "etcd.matchLabels" . | nindent 6 }}
+      etcd: headless
```

Comment:
Could you explain why this is needed?

Reply:
As-is, the serviceMonitor includes both the headless and non-headless service, and ultimately you end up with two sets of the same pods in the target list.

Reply:
What is the non-headless service used for? Maybe it would be better to allow it to be disabled.

Reply:
The non-headless service is used to access the etcd pods deployment from the clients

Reply:
Should I remove this? I think I can get by with relabelings, with mTLS/host authentication i need a complicated one already (not including pod IPs in my certificate):

```yaml
    relabelings:
    - separator: .
      sourceLabels:
      - __meta_kubernetes_pod_name
      - __meta_kubernetes_endpoints_name
      targetLabel: __pod_endpoint_address
    - separator: ':'
      sourceLabels:
      - __pod_endpoint_address
      - __meta_kubernetes_pod_annotation_prometheus_io_port
      targetLabel: __address__
```

Dropping the extra service isn't very hard, I just wasn't expecting both services to be targeted by the ServiceMonitor

Reply:
Here I'm preferring the headless service (only). My reasoning:
- When I view the targets list associated with this ServiceMonitor, I expect one target per pod.
- Without this change: I was seeing two targets per pod (one per each of the two services)

Reply:
I've reverted the `headless` service preference -- it diverges from the original intent of the PR.

Reply:
Thanks